app/webpacker/controllers/flatpickr_controller.js

```diff
@@ -60,6 +60,14 @@ export default class extends Flatpickr {
 
   open() {
     this.fp.element.dispatchEvent(new Event("focus"));
+    if (!this.defaultDateValue) return;
```

Comment:
I've another suggestion, that is, to me, easier to understand:
```js
  open() {
    this.fp.element.dispatchEvent(new Event("focus"));
    if (!this.fp.selectedDates.length) {
      this.setDefaultDateValue();
    }
  }

  ....
  // private
  setDefaultDateValue = () => {
    if (this.defaultDateValue === "startOfDay") {
      this.fp.setDate(moment().startOf("day").format());
    } else if (this.defaultDateValue === "endOfDay") {
      this.fp.setDate(moment().add(1, "days").startOf("day").format());
    }
  };
```
(not tested)
What do you think?

Reply:
I agree.
I tried to do that but the code was crashing because I didn't use `this` before `setDefaultDateValue`.

I'll follow this recommendation :)